### PR TITLE
[PageControl] Fix bug with OOB array lookup when calling scrollViewDidScroll.

### PR DESCRIPTION
**Pull request by kolodny**
The crash referenced in https://github.com/material-components/material-components-ios/pull/2072#issue-261964446 still applies even after the fix of #2132

This PR covers the invalid lookup in the `scrollViewDidScroll` method; tests included.

### CODE REVIEW

File: components/PageControl/src/MDCPageControl.m [hunk 1]
```diff
@@ -277,6 +277,11 @@ - (void)scrollViewDidScroll:(UIScrollView *)scrollView {
                    });
 
   } else if (scrolledPercentage >= 0 && scrolledPercentage <= 1) {
+
+    if (_numberOfPages == 0) {
```
Comment (ianegordon) [inline components/PageControl/src/MDCPageControl.m]: If this in a general guard statement, move to the beginning of the method?

If not, consider adding it in the if clause:
else if (scrolledPercentage >= 0 && scrolledPercentage <= 1 && _numberOfPages > 0)

A return in the middle of a method can easily get lost.

File: components/PageControl/src/MDCPageControl.m [hunk 2]
```diff
@@ -277,6 +277,11 @@ - (void)scrollViewDidScroll:(UIScrollView *)scrollView {
                    });
 
   } else if (scrolledPercentage >= 0 && scrolledPercentage <= 1) {
+
```
Comment (ianegordon) [inline components/PageControl/src/MDCPageControl.m]: Remove empty line?

**Review by ianegordon — CHANGES_REQUESTED** [review]

File: components/PageControl/src/MDCPageControl.m (hunk 2 above)
Comment (kolodny) [reply components/PageControl/src/MDCPageControl.m]: Done

File: components/PageControl/src/MDCPageControl.m (hunk 1 above)
Comment (kolodny) [reply components/PageControl/src/MDCPageControl.m]: Added it to the if clause above

**Review by ianegordon — APPROVED** [review]

**Comment by ianegordon** [comment]
Many thanks! I will merge once Travis is happy.

**Comment by romoore** [comment]
Closed #1015
Closed #2072

**Pull request merged**